src/gwt/src/org/rstudio/studio/client/common/JSONArrayBuilder.java

```diff
@@ -61,10 +61,15 @@ public final JSONArrayBuilder add(double value)
    
    public final JSONArrayBuilder add(JsArrayString value)
    {
+      if (value == null)
+         return appendNull();
+      
       JSONArray array = new JSONArray();
-      for (int i = 0, n = value.length(); i < n; i++)
-         array.set(i, fromString(value.get(i)));
+      if (value != null)
```

Comment:
Is this test superfluous? (looks like we'd have already have bailed if value was null)